teuthology/openstack/__init__.py

```diff
@@ -451,22 +477,33 @@ def __flavor_wrapper(self, min, good, hint, arch):
         This is the one, single place for coding OpenStack-provider-specific
         heuristics for selecting flavors.
         """
-        select = None
-        if self.get_provider() == 'ovh':
-            log.debug("Looking for a match among the El Cheapo flavors...")
-            select = '^vps-ssd-'
+        select_dict = {
+            #'ovh': ['^(s1|vps-ssd)-', '^(c2-[0-9]+|(hg|sg)-.*ssd)$', '^(hg|sg|c2)-.*ssd'],
+            'ovh': [
+                '^s1-', '^c2-[0-9]+$',          # new ovh flavors at first
+                '^vps-ssd-', '^(hg|sg)-.*ssd$'  # old ovh flavors
+            ],
+            'ecp': ['^(m1|m2).'],
+        }
+        if 'flavor' in teuth_config.openstack:
+            flavor_select = teuth_config.openstack['flavor'] or [None]
+        else:
+            flavor_select = select_dict[self.get_provider()] \
+                if self.get_provider() in select_dict else [None]
```

Comment:
the idea here is that one could force a particular set of regexes by setting "flavor" key in teuthology.conf ?

Reply:
That's correct.